### PR TITLE
[dualtor] add 'test_link_drop'

### LINKED ISSUE #2766: Standby ToR Link Drop Test Cases
Test all packet flow scenarios when the link between standby ToR and OVS bridge is dropping all packets

Depends on #2758, #2759

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Add the following tests to cover link drop scenarios:
1. test_active_link_drop_upstream
2. test_active_link_drop_downstream_active
3. test_active_link_drop_downstream_standby
4. test_standby_link_drop_upstream
5. test_standby_link_drop_downstream_active
6. test_standby_link_drop_downstream_standby

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

Summary:
Fixes #2766 #2765

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Cover the following test scenarios:
1. Test all packet flow scenarios when link between active ToR and OVS bridge is dropping all packets
2. Test all packet flow scenarios when link between standby ToR and OVS bridge is dropping all packets

#### How did you do it?
Add tests to cover the different traffic travel paths for the above two link drop scenarios.

#### How did you verify/test it?
Run `test_link_drop`

### CODE REVIEW

File: tests/dualtor/test_link_drop.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    return _set_drop_all_interfaces
+
+
+@pytest.fixture(scope="function")
+def drop_flow_upper_tor(set_drop, set_output, tor_mux_intfs):
+    """Drop the flow to the upper ToR."""
+    direction = "upper_tor"
+    yield _set_drop_factory(set_drop, direction, tor_mux_intfs)
+
+    for intf in tor_mux_intfs:
```
Comment (theasianpianist) [inline tests/dualtor/test_link_drop.py:36]: `set_drop` already cleans up after itself by restoring output, could you just change the scope to function and use it directly?

Comment (lolyu) [reply tests/dualtor/test_link_drop.py:36]: Nice catch, will update.

Comment (lolyu) [reply tests/dualtor/test_link_drop.py:36]: `set_drop` is a module-level fixture while the cleanup needs to be done for every test function, so let's keep this function-level restoration, and let `set_drop` be our last resort to do a double cleanup.

Comment (theasianpianist) [reply tests/dualtor/test_link_drop.py:36]: Sorry I meant change the scope of `set_drop` to function scope so it cleans up after every individual test. This way other tests that also use the fixture do not need to worry about function-level cleanup.

**Review by theasianpianist — APPROVED** [review]
lgtm, thanks!

**Pull request merged**